site/search/SearchPanel.tsx

```diff
@@ -0,0 +1,531 @@
+                            className={cx("search-filters__tab-button", {
+                                "search-filters__tab-button--is-active":
+                                    activeCategoryFilter === key,
+                            })}
+                        >
+                            {label}
+                            <span className="search-filters__tab-count">
+                                {hitsLengthByIndexName[key]}
+                            </span>
+                        </button>
+                    </li>
+                ))}
+            </ul>
+        </div>
+    )
+}
+
+function NoResultsBoundary({ children }: { children: React.ReactElement }) {
+    const { results } = useInstantSearch()
+
+    // The `__isArtificial` flag makes sure not to display the No Results message when no hits have been returned.
+    // Add the `hidden` attribute to the child <section> tag,
+    // which we can leverage along with the adjacent sibling selector
+    // to show a No Results screen with CSS alone
+    if (!results.__isArtificial && results.nbHits === 0) {
+        return React.cloneElement(children, { hidden: true })
+    }
+
+    return children
+}
+
+interface SearchResultsProps {
+    activeCategoryFilter: SearchCategoryFilter
+    isHidden: boolean
+    handleCategoryFilterClick: (x: SearchCategoryFilter) => void
+}
+
+const SearchResults = (props: SearchResultsProps) => {
+    const {
+        results: { queryID },
+    } = useInstantSearch()
+    const { activeCategoryFilter, isHidden, handleCategoryFilterClick } = props
+
+    // Listen to all clicks, if user clicks on a hit (and has consented to analytics - grep "hasClickAnalyticsConsent"),
+    // Extract the pertinent hit data from the HTML and log the click to Algolia
+    const handleHitClick = useCallback(
+        (event: MouseEvent) => {
+            if (!queryID) return
+            let target = event.target as HTMLElement | null
+            if (target) {
+                let isHit = false
+                while (target) {
+                    if (target.hasAttribute("data-algolia-object-id")) {
+                        isHit = true
+                        break
+                    }
+                    target = target.parentElement
+                }
+                if (isHit && target) {
+                    const objectId = target.getAttribute(
+                        "data-algolia-object-id"
+                    )
+                    const position = target.getAttribute(
+                        "data-algolia-position"
+                    )
+                    if (objectId && position) {
+                        logSiteSearchClick({
+                            index: SearchIndexName.Charts,
+                            queryID,
+                            objectIDs: [objectId],
+                            positions: [parseInt(position)],
+                        })
+                    }
+                }
+            }
+        },
```

Comment:
Hm, I haven't extensively checked this but there's definitely [a builtin `createInsightsMiddleware`](https://www.algolia.com/doc/api-reference/widgets/insights/js/) in instantsearch, maybe it's better to use that?